server/portal/apps/onboarding/steps/system_access_v3.py

```diff
@@ -12,29 +12,46 @@
 logger = logging.getLogger(__name__)
 
 
-def create_system_credentials(user, public_key, private_key, system_id, skipCredentialCheck=False) -> int:
+def create_system_credentials(client,
+                              username,
+                              public_key,
+                              private_key,
+                              system_id,
+                              skipCredentialCheck=False) -> int:
     """
     Set an RSA key pair as the user's auth credential on a Tapis system.
     """
-    logger.info(f"Creating user credential for {user.username} on Tapis system {system_id}")
+    logger.info(f"Creating user credential for {username} on Tapis system {system_id}")
     data = {'privateKey': private_key, 'publicKey': public_key}
-    client = user.tapis_oauth.client
+    client = client
     client.systems.createUserCredential(
         systemId=system_id,
-        userName=user.username,
+        userName=username,
         skipCredentialCheck=skipCredentialCheck,
         **data
     )
 
 
+def register_public_key(username, publicKey, system_id) -> int:
+    """
+    Push a public key to the Key Service API.
+    """
+    url = "https://api.tacc.utexas.edu/keys/v2/" + username
+    headers = {"Authorization": "Bearer {}".format(settings.KEY_SERVICE_TOKEN)}
+    data = {"key_value": publicKey, "tags": [{"name": "system", "value": system_id}]}
+    response = requests.post(url, json=data, headers=headers)
+    response.raise_for_status
+    return response.status_code
+
+
 def set_user_permissions(user, system_id):
     """Apply read/write/execute permissions to a user on a system."""
     logger.info(f"Adding {user.username} permissions to Tapis system {system_id}")
     client = service_account()
     client.systems.grantUserPerms(
         systemId=system_id,
         userName=user.username,
-        permissions=['READ', 'MODIFY', 'EXECUTE'])
+        permissions=['READ'])
```

Comment:
Good catch. We may want to change the string on line 48 as well, that currently reads: 

>     """Apply read/write/execute permissions to a user on a system."""